### PR TITLE
samples: nrf9160: modem_shell: Reduce heap config for Wi-Fi

### PR DESCRIPTION
**Pull request by trantanen**
CONFIG_HEAP_MEM_POOL_SIZE has been set to 100000 bytes as this was original value in samples/wifi/shell, which has now lowered heap to 25000.
MOSH needs a bit more for "location get" command purposes. It looks like one AP requires roughly 440 bytes with "location get" command. Peak heap usage with "kernel heap" command was
18044 bytes with 17 APs, and 27328 bytes with 38 APs. MOSH is now configured for 50 APs so calculation would give roughly 32000 bytes.
We'll reserve 40000 bytes which is way less than previous 100000.

Due to reduced heap usage, RAM is freed and we can add iPerf3, Curl, sock tool, SMS and FOTA into nrf7002ek configuration.

Setting also CONFIG_NRF700X_SCAN_LIMIT to same value of 50 as CONFIG_LOCATION_METHOD_WIFI_SCANNING_RESULTS_MAX_CNT.

Doing similar updates to location and nrf_cloud_mqtt_multi_service samples.

JIRA: MOSH-426

### CODE REVIEW

**Review by tokangas — APPROVED** [review]

**Comment by trantanen** [comment]
DNM until upmerge is done. I don't want to cause issues there and will rebase this PR when upmerge is completed.

File: samples/nrf9160/modem_shell/overlay-nrf7002ek-wifi-scan-only.conf
Comment (jhirsi) [inline samples/nrf9160/modem_shell/overlay-nrf7002ek-wifi-scan-only.conf:65]: I'd like to see similar comment here than is in PR describtion. I.e. to remind when we need to increase CONFIG_NRF700X_SCAN_LIMIT and thus also heap.

**Review by jhirsi — APPROVED** [review]

File: samples/nrf9160/modem_shell/overlay-nrf7002ek-wifi-scan-only.conf
Comment (trantanen) [reply samples/nrf9160/modem_shell/overlay-nrf7002ek-wifi-scan-only.conf]: Please check the comments.

**Review by glarsennordic — APPROVED** [review]
LGTM assuming MMSS+wifi still builds with these tweaks

Thanks for the additional configuration comments, those are genuinely appreciated

**Comment by trantanen** [comment]
All changed samples seems to work with 30-38 WiFi APs. Rebased the code still and waiting for CI to pass.

**Pull request merged**
